js/archive.js

```diff
@@ -0,0 +1,99 @@
+// Number of releases to show by default (rest will be behind a "show all" link).
+var releasesToShow = 5;
+
+// Fetches Flutter release JSON for the given OS and calls the callback once the data is available.
+var fetchFlutterReleases = function (os, callback, errorCallback) {
+  // OS: windows, macos, linux
+  var url = "https://storage.googleapis.com/flutter_infra/releases/releases_" + os + ".json";
+  $.ajax({
+    type: "GET",
+    url: url,
+    dataType: "json",
+    success: function (data) {
+      callback(data, os);
+    },
+    error: function (xhr, textStatus, errorThrown) {
+      errorCallback(os);
+    }
+  })
+}
+
+function convertReleaseFormat(releases, os) {
+  // Convert releases into a format that's easier to work with:
```

Comment:
If you want a different format in the source data, I can do that.  The website is the only consumer of this, so it's not going to break anyone.

Would you rather have the 'releases' member just be an array of dictionaries laid out like "New:" below?  I didn't do that because then I have to do duplicate checking when I add a new one to the list (to be sure that there isn't already one with the same hash/channel combination), but I can do that pretty easily.

Reply:
Your call, this code already works, but simplifying it is always good. The main reason I converted it was that it wasn't sorted (object keys don't have an order, so they basically come through sorted by the has in the json), but I then flattened it to simplify some other code.

If you want to change it, having an array pre-sorted by date would be good. Moving complexity to the server seems like a good idea IMO because we control the environment (the less code running in all the different browsers, the better!).

Reply:
By the way - we're currently pulling three files that are mostly the same, just with different archive links - would it be difficult/worthwhile to just have one file that has three archive links within each release object?

Reply:
No, we should leave them separate, because they are updated independently from three different build servers, and if they were one file, then it would be vulnerable to race conditions when updating the data (the windows build server could read the file, the linux one updates it, and then the windows one replaces it, reverting the linux update, for instance).  We could implement a locking scheme, but that would complicate the code, and I'd rather not do that until we have to.

Reply:
Ah, makes sense. I forgot this was more than just the git clone. LMK what you want to do about the format - if we're going to change it, doing it before we merge this is best!

Reply:
I am working on a PR that will generate what you asked for: I figure it'll be easier to do now before we're live.  I'll send it to you for review.

Reply:
My PR has been merged, and I've updated the existing metadata to the new format.